processes/vegetation_change.py

```diff
@@ -0,0 +1,350 @@
+        )
+
+        analysis_ds = dc.load(
+            time=analysis_time_period,
+            platform=platform_analysis,
+            product=analysis_product,
+            measurements=analysis_measurement,
+            **query,
+        )
+
+        if is_dataset_empty(baseline_ds):
+            raise Exception(
+                "DataCube Load returned an empty Dataset."
+                + "Please check load parameters for Baseline Dataset!"
+            )
+
+        if is_dataset_empty(analysis_ds):
+            raise Exception(
+                "DataCube Load returned an empty Dataset."
+                + "Please check load parameters for Analysis Dataset!"
+            )
+
+        baseline_clean_mask = mask_good_quality(baseline_ds, baseline_product)
+        analysis_clean_mask = mask_good_quality(analysis_ds, analysis_product)
+
+        xx_data_b = baseline_ds[all_measurements]
+        xx_data_a = analysis_ds[all_measurements]
+
+        baseline_ds_masked = odc.algo.keep_good_only(
+            xx_data_b, where=baseline_clean_mask
+        )
+        analysis_ds_masked = odc.algo.keep_good_only(
+            xx_data_a, where=analysis_clean_mask
+        )
+
+        if mosaic_type == "geomedian":
+            baseline_composite = geomedian(
+                baseline_ds_masked, baseline_product, all_measurements
+            )
+            analysis_composite = geomedian(
+                analysis_ds_masked, analysis_product, all_measurements
+            )
+        else:
+            mosaic_function = {
+                "median": create_median_mosaic,
+                "max": create_max_ndvi_mosaic,
+                "mean": create_mean_mosaic,
+            }
+            new_compositor = mosaic_function[mosaic_type]
+            baseline_composite = dask.delayed(new_compositor)(
+                baseline_ds_masked, clean_mask=baseline_clean_mask, no_data=np.nan
+            )
+            analysis_composite = dask.delayed(new_compositor)(
+                analysis_ds_masked, clean_mask=analysis_clean_mask, no_data=np.nan
+            )
+
+        if platform_base in ["LANDSAT_8", "LANDSAT_7", "LANDSAT_5", "LANDSAT_4"]:
+            water_scenes_baseline = dc.load(
+                product=baseline_water_product,
+                measurements=["water_classification"],
+                time=baseline_time_period,
+                **query,
+            )
+            water_scenes_baseline = water_scenes_baseline.where(
+                water_scenes_baseline >= 0
+            )
+            water_scenes_analysis = dc.load(
+                product=analysis_water_product,
+                measurements=["water_classification"],
+                time=analysis_time_period,
+                **query,
+            )
+            water_scenes_analysis = water_scenes_analysis.where(
+                water_scenes_analysis >= 0
+            )
+        else:
+            raise Exception("S2 does not yet have daskable water classification")

```

Comment:
We should not have the S2 option in the platform_base input if we are going to have this validation

Reply:
Good point. I've updated the PR